test/Interpreter/protocol_extensions.swift

```diff
@@ -76,7 +76,10 @@ extension MutableCollection
   where Self: RandomAccessCollection, Self.Iterator.Element : Comparable {
 
   public final mutating func myPartition() -> Index {
-    return self.partition()
+    if let first = self.first {
+      return self.partition(by: { $0 >= first})
+    }
+    return self.startIndex
```

Comment:
You can replace the body with just: `return self.partition(by: { $0 >= first! })`